### ISSUE TITLE
<inheritdoc /> with internal interfaces

### ISSUE THREAD

**Issue by danieltranca** [doxygen/doxygen#8866]
I have multiple classes that implement an internal interface. I want to use `<inheritdoc />` and get the docs from the interface since they all do the same thing.

According to [Sandcastle](https://ewsoftware.github.io/XMLCommentsGuide/html/86453FFB-B978-4A2A-9EB5-70E118CA8073.htm) this should be possible:
`For types and interfaces:

Inherit documentation from all base classes working backwards up the inheritance chain.

Inherit documentation from all interface implementations (if any) working through them in the order listed in the reflection information file (usually alphabetically).`
I can provide an example for this if needed(the repo in question is a public repo)

**Comment by albert-github** [comment]
- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.
- Please also specify the **full** doxygen version used (`doxygen -v`).

**Comment by danieltranca** [comment]
@albert-github I updated the ticket. Let me know if you need me to add any other info.

**Comment by albert-github** [comment]
I still don't see some essential information as requested in https://github.com/doxygen/doxygen/issues/8866#issuecomment-955658583:
- doxygen version used (`doxygen -v`)
- a tar / zip file with the code used and the settings used, I just see a snippet of some code. I think the problem might be in not setting some `EXTRACT_...` settings

**Comment by danieltranca** [comment]
@albert-github I have added a ss of the version I'm using and included a real example from the repo I'm applying the docs to. I also included the doxyfile that I actually use

**Comment by albert-github** [comment]
- the provided links are all moving targets and will, probably, in future have been changed.
- I **only** looked at the original example and used a fairly small Doxyfile, it just contains `EXTRACT_PACKAGE=YES` and in that case I do see the the information requested. From the documentation:
  > EXTRACT_PACKAGE If the EXTRACT_PACKAGE tag is set to YES, all members with package or internal scope
will be included in the documentation.

**Comment by danieltranca** [comment]
@albert-github I'm not looking to include the internal members but I'm looking to have their documentation added to the public member. I have set the EXTRACT_PACKAGE to YES and ran it, and the inheritdoc didn't work, but all the internal classes were added to the docs. I just want to mention that I do not want to include the internal members in the docs.

**Comment by danieltranca** [comment]
@albert-github I added Doxygen.Test.zip to the post. It's a unity project though...

**Comment by albert-github** [comment]
- I ran your test Doxyfile (from the directory Doxyge.Test/Doxygen after setting `EXTRACT_PACKAGE=YES` and I got:
![image](https://user-images.githubusercontent.com/5223533/139643094-e8076e2b-55e4-4a17-a093-b2dde36a6d6c.png)
  This looks like the documentation you want to accomplish (i.e. the brief description is present / inherited)
- the idea of documenting something, here the internal interface, but not adding it to the documentation and still using the documentation provided there is an interesting though I can imagine that this will give some problems with the, internal, structure of doxygen and with its philosophy (about what is available and what is not available).

**Comment by danieltranca** [comment]
@albert-github yes, this is exactly what I want. I have an interface that enforces the same behaviour on multiple classes and I don't want to document each behaviour, just the interface. But the interface should not be available for the library user. Is there any way this can be achieved or is it out of scope?

**Comment by albert-github** [comment]
The only thing I can think, at the moment, is to use a "snippet" with the documentation and use this on the different places.

Example: [snip.tar.gz](https://github.com/doxygen/doxygen/files/7459151/snip.tar.gz)

@doxygen what is your opinion about the general idea of documenting something, here the internal interface, but not adding it to the documentation and still using the documentation provided?

**Comment by danieltranca** [comment]
@albert-github @doxygen any news on this?

**Comment by danieltranca** [comment]
@albert-github  @doxygen  any news on this?

**Comment by RivenSkaye** [comment]
Friendly bump on this, as for me the C# form `<inheritdoc cref="BaseClass{Type1, Type2}.Method"` isn't working either (also in cases where there are no type args, and thus no curly braces)
